idioms/README.md

```diff
@@ -1,3 +1,7 @@
 # Idioms
 
-TODO: add description/explanation
+[Idioms](https://en.wikipedia.org/wiki/Programming_idiom) are commonly used styles and patterns largely agreed upon by a community. They are guidelines. Writing idiomatic code allows other developers to understand what is happening because they are familiar with the form that it has.
+
+The computer understands the machine code that is generated by the compiler. The language is therefore mostly beneficial to the developer. Since we have this abstraction layer, why not put it to good use and make it simple? Remember the KISS principle.
```

Comment:
```suggestion
The computer understands the machine code that is generated by the compiler. The language is therefore mostly beneficial to the developer. So, since we have this abstraction layer, why not put it to good use and make it simple?
Remember the KISS principle: "Keep It Simple, Stupid".
```

Reply:
Maybe I would totally delete the phrase "Remember the KISS principle: "Keep It Simple, Stupid".
I like the kiss principle, but that single phrase seems a little bit out of context here. What do you think?

Reply:
Maybe that gives people that are not familiar with the KISS principle some small input to Google it and read it up themselves? Not sure though. Maybe another small phrase describing why it's important like an explanatory cliffhanger could be good then?

Reply:
Added a small claim of the KISS principle to make clearer. What do you think, still to be removed?

Reply:
It's really nice now, thanks!